jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java

```diff
@@ -258,28 +260,34 @@ else if (link.isThrowLink()) {
         // throw exception if any error (this is done at the end of the process to show the user as much errors as possible) 
         StringBuilder errors = new StringBuilder();
         if (!noNameLinks.isEmpty()) {
-            formatError(errors,"These nodes do not have a name ",noNameLinks.stream());
+            formatError(errors,"These nodes do not have a name ",noNameLinks.stream(), process);
         }
         if (!duplicatedTarget.isEmpty()) {
-            formatError(errors,"\nThere are multiple catch nodes with the same name ",duplicatedTarget.stream());
+            formatError(errors,"\nThere are multiple catch nodes with the same name ",duplicatedTarget.stream(), process);
         }
         if (!unconnectedTarget.isEmpty()) {
-            formatError(errors,"\nThere is not connection from any throw link to these catch links ",unconnectedTarget.stream());
+            formatError(errors,"\nThere is not connection from any throw link to these catch links ",unconnectedTarget.stream(), process);
         }
         if (!throwLinks.isEmpty()) {
             formatError(errors,"\nThere is not connection from any catch link to these throw links ",throwLinks
                                                .values()
                                                .stream()
-                                               .flatMap(Collection::stream));
+                                               .flatMap(Collection::stream), process);
         }
         if (errors.length() > 0) {
             throw new IllegalArgumentException (errors.toString());
         }
         
     }
     
-    private static void formatError (StringBuilder errors, String message, Stream<IntermediateLink> stream) {
+    private static void formatError (StringBuilder errors, String message, Stream<IntermediateLink> stream, NodeContainer container) {
         errors.append(message).append(stream.map(IntermediateLink::getUniqueId).collect(Collectors.joining(", ","{","}")));
+        if (container instanceof Process) {
```

Comment:
Maybe we can use `org.kie.api.definition.process.Process` instead, just to make it a little more abstract? just like the one used below which is an instance of `org.kie.api.definition.process.Node`? - wdyt?

Reply:
you are right, changed